pkg/vsphere/vm/vm.go

```diff
@@ -560,3 +564,41 @@ func (vm *VirtualMachine) DatastoreReference(ctx context.Context) ([]types.Manag
 	}
 	return mvm.Datastore, nil
 }
+
+// GetVCHUpdateStatus tells if an upgrade/update has already been started based on the UpdateInProgress flag in ExtraConfig
+// If error != nil, VCH upgrade/update will exit; so the value of the VCH upgrade/update status is not important
+func (vm *VirtualMachine) GetVCHUpdateStatus(ctx context.Context) (bool, error) {
+	info, err := vm.FetchExtraConfig(ctx)
+	if err != nil {
+		log.Errorf("Unable to get vm ExtraConfig: %s", err)
+		return false, err
+	}
+
+	if v, ok := info[UpdateInProgress]; ok {
+		updateStatus, err := strconv.ParseBool(v)
+		if err != nil {
+			//  If error occurs, return true to cancel VCH upgrade. The user can reset the UpdateInProgress flag later.
```

Comment:
if error happens, the bool return value should not matter for the caller. 
again do not comment caller's behavior